fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientIdentifierReadPlatformServiceImpl.java

```diff
@@ -103,7 +106,14 @@ public ClientIdentifierData mapRow(final ResultSet rs, @SuppressWarnings("unused
             final String documentTypeName = rs.getString("documentType");
             final CodeValueData documentType = CodeValueData.instance(documentTypeId, documentTypeName);
             final String status = ClientIdentifierStatus.fromInt(rs.getInt("status")).getCode();
-            return ClientIdentifierData.singleItem(id, clientId, documentType, documentKey, status, description);
+
+            String documentIssueCountry = null;
+            if (documentTypeName.equals(ClientApiConstants.clientIdentifierPassportParamName)) {
+                Map<String, String> translateCountryList = CountryNamesTranslationService.translateCountryList();
+                documentIssueCountry = translateCountryList.get(rs.getString("documentIssueCountryCode"));
```

Comment:
Yeah this is exactly what I meant by the interface suggestion above. The client preferably should only tell the service "I want a resolved country name for this country code" and it shouldn't play around the collection if it doesn't need all of them.